src/components/views/auth/InteractiveAuthEntryComponents.tsx

```diff
@@ -672,6 +672,92 @@ export class MsisdnAuthEntry extends React.Component<IMsisdnAuthEntryProps, IMsi
     }
 }
 
+interface IRegistrationTokenAuthEntryState {
+    registrationToken: string;
+}
+
+export class RegistrationTokenAuthEntry extends React.Component<IAuthEntryProps, IRegistrationTokenAuthEntryState> {
+    public static readonly LOGIN_TYPE = AuthType.RegistrationToken;
+
+    public constructor(props: IAuthEntryProps) {
+        super(props);
+
+        this.state = {
+            registrationToken: null,
+        };
+    }
+
+    public componentDidMount() {
+        this.props.onPhaseChange(DEFAULT_PHASE);
+    }
+
+    private onSubmit = (e: FormEvent) => {
+        e.preventDefault();
+        if (this.props.busy) return;
+
+        this.props.submitAuthDict({
+            // Could be AuthType.RegistrationToken or AuthType.UnstableRegistrationToken
+            type: this.props.loginType,
+            token: this.state.registrationToken,
+        });
+    };
+
+    private onRegistrationTokenFieldChange = (ev: ChangeEvent<HTMLInputElement>) => {
+        // enable the submit button if the registration token is non-empty
+        this.setState({
+            registrationToken: ev.target.value,
+        });
+    };
+
+    public render() {
+        const registrationTokenBoxClass = classNames({
+            "error": this.props.errorText,
+        });
+
+        let submitButtonOrSpinner;
+        if (this.props.busy) {
+            submitButtonOrSpinner = <Spinner />;
+        } else {
+            submitButtonOrSpinner = (
+                <AccessibleButton
+                    onClick={this.onSubmit}
+                    kind={this.props.continueKind || 'primary'}
```

Comment:
looks like the linter is complaining that the property doesn't exist - we might need to create an interstitial interface which extends `IAuthEntryProps` to use in the `React.Component<>` type instead.

Reply:
likewise with `continueText` below

Reply:
Ah, thanks for taking a look, I did notice that too when I updated it to develop and just remived those properties (don't know if that's OK?), but I forgot to push!